python/create_service_diagnostics_bundle.py

```diff
@@ -226,11 +236,12 @@ def main(argv) -> int:
             return 0
 
     rc, _ = FullBundle(
-        args.get("package_name"), args.get("service_name"), args.get("bundles_directory")
+        args.get("package_name"), args.get("service_name"), args.get("bundles_directory"),
+        dcos_version, args.get("service_diagnostics_version")
```

Comment:
The parameter list is getting rather long, and the `.get()` calls are not pretty either, and any typo there will escape static checks.
Can you please refactor `preflight_check` to return a custom first-class object rather than a dict?

Reply:
Done